assets/arc-0031/client/utils/hooks/useAuth.ts

```diff
@@ -51,15 +55,17 @@ export const useAuth = () => {
     showErrorNotification(normalizeError(error))
   }
 
+  const prefix = 'arc0031'
+
   const signIn = async () => {
     try {
       await connectWallet()
-      const message = await new Arc31ApiClient().request(address.value)
-      // Remove the 'AX' prefix and decode the message from base64
-      const decodedMessage = decode(Buffer.from(message.replace(/^AX/, ''), 'base64'))
-      // Extract the authMessage from the decoded message
-      const authMessageWrapper = decodedMessage as { 'arc31:j': AuthMessage }
-      authMessage.value = authMessageWrapper['arc31:j']
+      const chainId = env.client.algorandChainId
+      const message = await new Arc31ApiClient().request(address.value, chainId.toString())
+      // Remove the prefix from the message
+      const decodedMessage = message.slice(prefix.length);
```

Comment:
What if the message prefix has the same length of `prefix` but different content, i.e. `arc0000`? Shouldn't we simply check for the prefix to be `arc0031` and throw an error if that's not the case?

Reply:
Good catch! I have added the check here https://github.com/deanstef/ARCs/pull/3/commits/9fea035fea3f3871a845553e0098d2a8bf531040

Also, there was a bug in the error handling, fixed here: https://github.com/deanstef/ARCs/commit/9fea035fea3f3871a845553e0098d2a8bf531040#diff-b6585e3ad4caa3007d092451cad422c24442e0db925bc0c26213aa60acb210faR74